spring-integration-jdbc/src/main/java/org/springframework/integration/jdbc/store/JdbcMessageStore.java

```diff
@@ -105,12 +103,9 @@
 
 		COUNT_ALL_MESSAGES_IN_GROUP("SELECT COUNT(MESSAGE_ID) from %PREFIX%GROUP_TO_MESSAGE where GROUP_KEY=? and REGION=?"),
 
-		LIST_MESSAGEIDS_BY_GROUP_KEY("select MESSAGE_ID, CREATED_DATE " +
-				"from %PREFIX%MESSAGE where MESSAGE_ID in (select MESSAGE_ID from %PREFIX%GROUP_TO_MESSAGE where GROUP_KEY=? and REGION=?) " +
-				"ORDER BY CREATED_DATE"),
-
 		LIST_MESSAGES_BY_GROUP_KEY("SELECT MESSAGE_ID, MESSAGE_BYTES, CREATED_DATE " +
-				"from %PREFIX%MESSAGE where MESSAGE_ID in (SELECT MESSAGE_ID from %PREFIX%GROUP_TO_MESSAGE where GROUP_KEY = ?) and REGION=? " +
+				"from %PREFIX%MESSAGE where MESSAGE_ID in " +
+				"(SELECT MESSAGE_ID from %PREFIX%GROUP_TO_MESSAGE where GROUP_KEY = ? and REGION = ?) and REGION = ? " +
```

Comment:
Why do we need REGION twice? Sure the `in` clause will only have ids in that region so the and part will be a no-op.

Reply:
That's not true. Primary key is:
```
constraint MESSAGE_PK primary key (MESSAGE_ID, REGION)
```
So, definitely the same message may be in different regions and, therefore, selecting just ids for messages is not enough.

Let me know what ma I missing.